examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,392 @@
+pooling rules. This applies to an estimate like explained variance.
+
+In sum, Rubin's pooling rules are as follows. The overall point estimate after
+multiple imputation (denoted by Qbar) is the average of all the m point
+estimates. The variance of the overall point estimate is a combination of
+so-called within imputation variance (Ubar) and between imputation
+variance (B). Ubar is the average of the m variances of the m point estimates.
+Both Qbar and Ubar are corrected with a factor 1 / m to account for sampling
+variance. The between imputation variance (B) is the sum of the squared
+difference between Qbar and the m point estimates, corrected with a factor
+1 / (m – 1). Then, the total variance (T) of the MI overall point estimate is
+Ubar + B + B/m.
+
+In this Example we will show how to use the `sklearn.impute.IterativeImputer`
+to perform multiple imputation. In Example 1 we show the effect of Rubin’s
+pooling rules on the variance of regression estimates. Due to the between
+imputation variance, the standard errors of all regression coefficients are
+larger with multiple imputation than with single imputation.
+
+In Example 2 we show how to set up a prediction model using multiple
+imputation. We combine the predictions from multiple imputations and show that
+this results in better regression performance.
+
+# Authors: Rianne Schouten <r.m.schouten@uu.nl>
+           Sergey Feldman <sergeyfeldman@gmail.com>
+# License: BSD 3 clause
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats, special
+
+from sklearn.datasets import load_boston
+from sklearn.linear_model import RidgeCV
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.impute import IterativeImputer
+from sklearn.metrics import mean_squared_error as mse
+
+rng = np.random.RandomState(0)
+
+N_ITER = 20  # number of iterations in IterativeImputer
+N_SIM = 3  # number of simulations in Example 2
+
+
+def ampute(X, missing_rate=0.75, mech="MCAR"):
+    """Ampute (opposite of impute) function to simulate missing data in X
+
+    MCAR - missing completely at random
+    MNCAR - missing not completely at random
+    """
+    X_out = X.copy()
+    n_drop_per_feature = int(missing_rate * X.shape[0])
+    for x in np.transpose(X_out):
+        # insert missing values for each feature
+        if mech == 'MCAR':
+            prob = None
+        else:
+            weights = special.expit(x - x.mean())
+            prob = weights / weights.sum()
+        drop_idx = np.random.choice(X.shape[0], p=prob, replace=False,
+                                    size=n_drop_per_feature)
+        x[drop_idx] = np.nan
+    return X_out
+
+
+# Make a function that calculates the variance of the beta estimates.
+def calculate_variance_of_beta_estimates(y_true, y_pred, X):
+    sum_sq_errors = np.sum((y_true - y_pred)**2)
+    sigma_hat_squared = sum_sq_errors / (len(y_true) - 2)
```

Comment:
By curiosity, why do we need `- 2`?

Reply:
https://stats.stackexchange.com/questions/204238/why-divide-rss-by-n-2-to-get-rse